platform/broadcom/sonic-platform-modules-dell/s5296f/sonic_platform/sfp.py

```diff
@@ -0,0 +1,372 @@
+]
+QSFP_TYPE_LIST = [
+    '0xc', # QSFP
+    '0xd', # QSFP+ or later
+    '0x11'  # QSFP28 or later
+]
+QSFP_DD_TYPE_LIST = [
+    '0x18' #QSFP-DD Type
+]
+
+class Sfp(SfpOptoeBase):
+    """
+    DELLEMC Platform-specific Sfp class
+    """
+    BASE_RES_PATH = "/sys/bus/pci/devices/0000:04:00.0/resource0"
+
+    def __init__(self, index, sfp_type, eeprom_path):
+        SfpOptoeBase.__init__(self)
+        self.port_type = sfp_type
+        self.sfp_type = sfp_type
+        self.index = index
+        self.eeprom_path = eeprom_path
+        self._initialize_media(delay=False)
+
+    def get_eeprom_path(self):
+        return self.eeprom_path
+
+    def get_name(self):
+        return "SFP/SFP+/SFP28" if self.index < 96 else "QSFP28 or later"
+
+    def pci_mem_read(self, mm, offset):
+        mm.seek(offset)
+        read_data_stream = mm.read(4)
+        reg_val = struct.unpack('I', read_data_stream)
+        mem_val = str(reg_val)[1:-2]
+        # print "reg_val read:%x"%reg_val
+        return mem_val
+
+    def pci_mem_write(self, mm, offset, data):
+        mm.seek(offset)
+        # print "data to write:%x"%data
+        mm.write(struct.pack('I', data))
+
+    def pci_set_value(self, resource, val, offset):
+        fd = os.open(resource, os.O_RDWR)
+        mm = mmap.mmap(fd, 0)
+        val = self.pci_mem_write(mm, offset, val)
+        mm.close()
+        os.close(fd)
+        return val
+
+    def pci_get_value(self, resource, offset):
+        fd = os.open(resource, os.O_RDWR)
+        mm = mmap.mmap(fd, 0)
+        val = self.pci_mem_read(mm, offset)
+        mm.close()
+        os.close(fd)
+        return val
+
+    def _initialize_media(self,delay=False):
+        """
+        Initialize the media type and eeprom driver for SFP
+        """
+        if delay:
+            time.sleep(1)
+            self._xcvr_api = None
+            self.get_xcvr_api()
+
+        self.set_media_type()
+        self.reinit_sfp_driver()
+
+    def get_position_in_parent(self):
+        """
+        Retrieves 1-based relative physical position in parent device.
+        Returns:
+            integer: The 1-based relative physical position in parent
+            device or -1 if cannot determine the position
+        """
+        return self.index
+
+    def is_replaceable(self):
+        """
+        Indicate whether this device  is replaceable.
+        Returns:
+            bool: True if it is replaceable.
+        """
+        return True
+
+    def get_presence(self):
+        """
+        Retrieves the presence of the sfp
+        Returns : True if sfp is present and false if it is absent
+        """
+        # Check for invalid port_num
+        mask = {'QSFP' : (1 << 4), 'SFP' : (1 << 0)}
+        # Port offset starts with 0x4004
+        port_offset = 16388 + ((self.index-1) * 16)
+
+        try:
+            status = self.pci_get_value(self.BASE_RES_PATH, port_offset)
+            reg_value = int(status)
```

Comment:
Status is not an appropriate variable name here.

Reply:
Updated.